src/erldns_zone_parser.erl

```diff
@@ -753,7 +769,8 @@ json_record_to_erlang([Name, Type = <<"DNSKEY">>, Ttl, Data, _Context]) when is_
                         flags = maps:get(<<"flags">>, Data),
                         protocol = maps:get(<<"protocol">>, Data),
                         alg = maps:get(<<"alg">>, Data),
-                        public_key = PublicKey
+                        public_key = PublicKey,
+                        key_tag = 0
```

Comment:
Where is this change originating? Is it anyhow changing the current behavior?

Reply:
It comes from the new type requirements, requiring this element to be an integer. This is read when decoding, but in code it is never used during encoding as far as I could find. We can allow it to be an undefined if we update dns_erlang, but it seems to me from RFC4034 (https://datatracker.ietf.org/doc/html/rfc4034#section-3.1.6) that this is never supposed not to be an integer 🤔